.github/workflows/nameguard-end-to-end-tests.yml

```diff
@@ -0,0 +1,91 @@
+name: NameGuard End to End Tests
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "packages/nameguard-sdk/**"
+      - "api/nameguard/**"
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "packages/nameguard-sdk/**"
+      - "api/nameguard/**"
+
+jobs:
+  check_changes:
+    runs-on: ubuntu-latest
+    outputs:
+      api_changed: ${{ steps.check.outputs.api_changed }}
+      sdk_changed: ${{ steps.check.outputs.sdk_changed }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check for changes
+        id: check
+        run: |
+          git fetch origin ${{ github.base_ref }}
+          API_CHANGED=$(git diff --name-only origin/${{ github.base_ref }} HEAD | grep "^api/nameguard/" | wc -l)
+          SDK_CHANGED=$(git diff --name-only origin/${{ github.base_ref }} HEAD | grep "^packages/nameguard-sdk/" | wc -l)
+          echo "api_changed=$API_CHANGED" >> $GITHUB_OUTPUT
+          echo "sdk_changed=$SDK_CHANGED" >> $GITHUB_OUTPUT
+
+  test:
+    needs: check_changes
+    if: needs.check_changes.outputs.api_changed > 0 || needs.check_changes.outputs.sdk_changed > 0
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "lts/*"
+          cache: "pnpm"
+
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v3
+        with:
+          version: 8
+          run_install: true
+
+      - name: Install poetry
+        if: needs.check_changes.outputs.api_changed > 0
+        run: pipx install poetry
+
+      - name: Set up Python
+        if: needs.check_changes.outputs.api_changed > 0
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+          cache: "poetry"
+
+      - name: Install dependencies
+        if: needs.check_changes.outputs.api_changed > 0
+        working-directory: ./api
+        run: poetry install --with dev
+
+      - name: Build the SDK
+        run: pnpm build
+
+      - name: Setup local nameguard api
+        if: needs.check_changes.outputs.api_changed > 0
+        working-directory: ./api
+        env:
+          PROVIDER_URI_MAINNET: ${{ secrets.PROVIDER_URI_MAINNET }}
+          PROVIDER_URI_SEPOLIA: ${{ secrets.PROVIDER_URI_SEPOLIA }}
+        run: |
+          poetry run uvicorn nameguard.web_api:app --host 0.0.0.0 --port 8000 &
+          sleep 5
+
+      - name: Local api healthcheck
+        if: needs.check_changes.outputs.api_changed > 0
+        run: curl http://0.0.0.0:8000/v1-beta/inspect-name/mainnet/test.eth
+
+      - name: Run tests
+        env:
+          PROVIDER_URI_MAINNET: ${{ secrets.PROVIDER_URI_MAINNET }}
+          PROVIDER_URI_SEPOLIA: ${{ secrets.PROVIDER_URI_SEPOLIA }}
+          NAMEGUARD_TEST_ENDPOINT: ${{ needs.check_changes.outputs.api_changed > 0 && 'http://0.0.0.0:8000' || '' }}
```

Comment:
@lightwalker-eth @notrab Here, if API doesn't change then we test SDK against production endpoint - is it ok or should test against local endpoint?
Not running local endpoint might be faster.
Production and local endpoints might give different results because production endpoint is not automatically deployed - so I think it should be tested against local version.

Reply:
However, nameguard-sdk is automatically released (right?) so there might be a mismatch.

Reply:
`nameguard-sdk` is released on merge to main, so tests would have to have been approved prior to that, am I right?

Reply:
So if we change SDK and API then we are testing SDK against local API.
So test against local API pass, and test against prod API fail. If we skip test against prod API, then released SDK will not work (against prod API).

Reply:
Ok, we can revert

Reply:
There is no solution for this.

Reply:
imo, we can always test local running api + sdk when api and/or sdk changes.

We have to assume what ends up in prod is good, and passing tests.

We could alternatively, extend the deploy api workflow to run the SDK tests there again to be safe, but it sounds like overkill.